src/misc_early.rs

```diff
@@ -113,4 +126,27 @@ impl EarlyLintPass for MiscEarly {
             }
         }
     }
+
+    fn check_expr(&mut self, cx: &EarlyContext, expr: &Expr) {
+        if let ExprKind::Call(ref paren, _) = expr.node {
+            if let ExprKind::Paren(ref closure) = paren.node {
+                if let ExprKind::Closure(_, _, _) = closure.node {
+                    cx.span_lint(CLOSURE_CALLED_EARLY, expr.span, "Try not to call \
+                        a closure in the expression where it is declared.")
+                }
+            }
+        }
+    }
+
+    fn check_block(&mut self, cx: &EarlyContext, block: &Block) {
+        for w in block.stmts.windows(1) {
+            if let &StmtKind::Decl(ref item,_) = &w[0].node {
+                if let DeclKind::Local(ref closure) = item.node {
+                    if let ExprKind::Closure(_, _, _) = closure.node {
```

Comment:
Is this correct? By [https://manishearth.github.io/rust-internals-docs/syntax/ast/enum.DeclKind.html#variant.Local](url) I guess I'm supposed to look for `closure.ty.node` but the compiler screams when I do so